lib/resources/datasets.js

```diff
@@ -51,6 +52,28 @@ module.exports = (service, endpoint) => {
     return Datasets.getMetadata(updatedDataset);
   }));
 
+  service.post('/projects/:id/datasets', endpoint(async ({ Projects, Datasets }, { auth, body, params }) => {
+    const project = await Projects.getById(params.id).then(getOrNotFound);
+    await auth.canOrReject('dataset.create', project);
+
+    const { name } = body;
```

Comment:
I was going down this route but it felt uncomfortable to make `name` writable, because then the name of a dataset could be changed via the PATCH request! I guess if I get `name` straight from the body, only in this POST/create resource, and then use `fromApi()` for other things like `approvalRequired`, that's okay.

Reply:
Yeah, it's definitely an unusual situation! Not sure if there's a better approach. I agree that `name` shouldn't be writable.